packages/adapters-api/src/build-api.ts

```diff
@@ -1,131 +1,237 @@
+import { OpenAPIHono, createRoute, z } from '@hono/zod-openapi'
 import { DefiProvider } from '@metamask-institutional/defi-adapters'
-import { Hono } from 'hono'
+import { type PinoLogger, pinoLogger } from 'hono-pino'
 import { cors } from 'hono/cors'
-import { ZodError } from 'zod'
 import './bigint-json.js'
+import { type RequestIdVariables, requestId } from 'hono/request-id'
+import type { ContentfulStatusCode } from 'hono/utils/http-status'
 import type { DbService } from './db-service.js'
 import {
-  GetPositionsSchema,
-  GetSupportSchema,
+  INTERNAL_SERVER_ERROR,
+  OK,
+  UNPROCESSABLE_ENTITY,
+} from './http-status-codes.js'
+import { logger } from './logger.js'
+import {
+  GetPositionsParamsSchema,
+  GetPositionsQuerySchema,
+  GetPositionsResponseSchema,
+  GetSupportQuerySchema,
+  GetSupportResponseSchema,
   IsEthAddress,
+  ValidationErrorSchema,
 } from './schemas.js'
 
-export function buildApi(
-  app: Hono,
-  defiProvider: DefiProvider,
-  dbService: DbService,
-) {
+function createApp() {
+  return new OpenAPIHono<{
+    Variables: RequestIdVariables & {
+      logger: PinoLogger
+    }
+  }>({
+    strict: false,
+    defaultHook: (result, c) => {
+      if (!result.success) {
+        return c.json(
+          {
+            success: result.success,
+            error: result.error,
+          },
+          UNPROCESSABLE_ENTITY,
+        )
+      }
+    },
+  })
+}
+
+function addMiddleware(app: ReturnType<typeof createApp>) {
   app.use('*', cors())
+  app.use('*', requestId())
+  app.use(
+    '*',
+    pinoLogger({
+      pino: logger,
+    }),
+  )
 
-  app.get('/health', (context) => context.json({ message: 'ok' }))
+  app.onError((err, c) => {
```

Comment:
This formats the error response and status code if we throw from within the API.